nuxt3-pinia-vuetify/hooks/useFetchApI.ts

```diff
@@ -0,0 +1,13 @@
+import { UseFetchOptions } from "nuxt/app";
+
+export default function useFetchAPI(url: string, opts?: UseFetchOptions<unknown>) {
```

Comment:
This is just a suggestion, but what about specifying a generic return type for the hook?

```typescript
export default function useFetchAPI<DataT>(url: string, opts?: UseFetchOptions<DataT>) {
const config = useRuntimeConfig();
	const { GITHUB_API_URL } = config;
	const token = useCookie('token');
	const headers = {
		...(opts?.headers || {}),
		...(token && { Authorization: `Bearer ${token.value}` }),
	};

	return useFetch(url, { ...opts, headers, baseURL: GITHUB_API_URL });
}
```

So that you avoid casting the return type (using `as`)
```typescript
const { data } = await useFetchAPI<IUser>(url, {
					headers: {
						Accept: 'application/vnd.github+json',
					},
				});
```